OAK-10341

nfsantos opened the pull request.

Commit:
OAK-9922 Parallel Compaction

Commit:
Make SegmentBufferWriterPool implementation a configurable option

Commit:
Address review comments

Commit:
Address further comments

thomasmueller committed:
OAK-10333 Improved logging for queries that that traverse more than 10'000 nodes

mreutegg committed:
OAK-10340: Child order not updated on source after move

Test to reproduce the issue

mreutegg committed:
OAK-10340: Child order not updated on source after move

Ensure child order on old and new parent

reschke committed:
OAK-10342: Update Apache Parent POM to version 30

reschke committed:
OAK-10338: Change test to tolerate different line endings

kwin committed:
OAK-10327 Embedded dependencies should have "provided" scope (#1004)

Otherwise they are transitively inherited by depending Maven modules

reschke committed:
OAK-10326: remove baseline comparison version (back to default)

reschke committed:
Revert "OAK-10342: Update Apache Parent POM to version 30"

This reverts commit 288e81cacbd3af6a76aa66a9ccfce2e01d9fbeac.

nfsantos committed:
Modify pipelined strategy to create a TreeStore instead of a FFS. (TESTING).

nfsantos committed:
Add missing file from previous commit.

mreutegg committed:
OAK-10340: Child order not updated on source after move

Additional test that verifies Tree parent stays the same when move fails at NodeBuider.moveTo()

nfsantos committed:
Merge remote-tracking branch 'upstream/OAK-10341' into OAK-10341-nuno

kwin committed:
OAK-10332 Allow reusing FilterProviderImpl outside OSGi context (#1014)

nfsantos committed:
Separate implementations of pipeline strategy with tree store and with FFS.
Add options to select using tree store or FFS.

nfsantos committed:
Do not serialize node state entries when passing them from transform to tree-store stages.

thomasmueller commented on oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/TreeSortPipelinedTransformTask.java:
```diff
@@ -0,0 +1,232 @@
+                                        nseBatch = new ArrayList<>(MAX_NSE_BATCH_SIZE);
+                                    }
+                                    // Write entry to buffer
+                                    nseBatch.add(new NodeStateEntryJson(nse.getPath(), entryWriter.asJson(nse.getNodeState())));
```
here we need to add the entry for the child node name... that would be something like this (not compiled or tested):
```
String path = nse.getPath();
if (!path.equals("/")) {
    String nodeName = PathUtils.getName(path);
    String parentPath = PathUtils.getParentPath(path);
    nseBatch.add(new NodeStateEntryJson(parentPath + "\t" + nodeName, ""));
}
```

nfsantos committed:
Merge remote-tracking branch 'upstream/OAK-10341' into OAK-10341-nuno

nfsantos committed:
Merge remote-tracking branch 'upstream/OAK-10341' into OAK-10341-nuno

mbaedke committed:
OAK-10349: Update Tomcat dependency to 9.0.78 (#1022)

nfsantos committed:
Experiments.

thomasmueller committed:
OAK-10341 - less merging

thomasmueller committed:
OAK-10341 - less merging: logging

ahanikel committed:
OAK-10345 - Add debug log message if larger blob is compared byte-wise. (#1024)

* OAK-10345 - Add debug log message if larger blob is compared byte-wise.

* OAK-10345 - Add an enable switch to make this as cheap as possible.

* OAK-10345 - Use SystemPropertySupplier for system properties.

---------

Co-authored-by: Axel Hanikel <ahanikel@adobe.com>

reschke committed:
OAK-10352: Update (shaded) Guava to 32.1.1 (#1028)

mreutegg committed:
Merge branch 'trunk' into OAK-10340

mreutegg committed:
Merge pull request #1017 from mreutegg/OAK-10340

OAK-10340: Child order not updated on source after move

reschke committed:
OAK-10354: Release Oak 1.54.0 - Candidate Release Notes (#1030)

tihom88 committed:
OAK-10273: Index-definition json created during oak-run reindex shoul… (#991)

* OAK-10273: Index-definition json created during oak-run reindex should not serialise index data

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.54.0

reschke committed:
[maven-release-plugin] rollback the release of jackrabbit-oak-1.54.0

reschke committed:
wq
Merge branch 'trunk' of https://gitbox.apache.org/repos/asf/jackrabbit-oak into trunk

reschke committed:
OAK-10354: Release Oak 1.54.0 - update release notes

reschke committed:
Merge branch 'trunk' of https://gitbox.apache.org/repos/asf/jackrabbit-oak into trunk

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.54.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

mreutegg committed:
OAK-10357: Path option for documentstore-check

nfsantos committed:
OAK-10356 - Adjust lower and upper bounds of auto-detected memory limits in PipelinedStrategy (#1032)

* Reduce the minimum bound and set a maximum bound for the auto-detected working memory in the Pipelined strategy.

* Reduce default number of transform threads.

mbaedke committed:
OAK-10350: Update spring-boot dependency to version 2.7.13 (#1027)

Co-authored-by: Manfred Baedke <gre55877@adobe.com>

reschke committed:
Revert "[maven-release-plugin] prepare for next development iteration"

This reverts commit cab87a64fe5898274febde2c94e60fe27cbb97d2.

reschke committed:
Revert "[maven-release-plugin] prepare release jackrabbit-oak-1.54.0"

This reverts commit 5e3048b9ae04bda853086f7a2c7d1880ef96226f.

mbaedke committed:
OAK-10361: Exclude dependency-reduced-pom.xml from the source package during release (#1033)

Co-authored-by: Manfred Baedke <gre55877@adobe.com>

reschke committed:
OAK-10354: Release Oak 1.54.0 - update release notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.54.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

fabriziofortino committed:
OAK-10353: Elastic custom analyzer should ignore unsupported properties (#1029)

* OAK-10353: Elastic custom analyzer should ignore unsupported properties (eg: hidden, arrays)

* OAK-10353: (doc) add link to lucene logic for unsupported props

* test: fix Lowecase -> LowerCase

anchela committed:
OAK-10362: Create benchmark test (subtask of Improve AutoMembershipPrincipals#isInheritedMember)

anchela committed:
OAK-10364 : Consolidate inheritance related automembership tests (subtask of OAK-10318 Improve AutoMembershipPrincipals#isInheritedMember)

reschke committed:
OAK-10274: oak-core-spi: remove Guava from public API (#982)

* OAK-10247: oak-commons: remove Guava from public API

* OAK-10247: temp. switch project version for testing

* Revert "OAK-10247: temp. switch project version for testing"

This reverts commit 3994dc9e84b5d2773f7d7dd1f2d0333806e93e88.

* OAK-10247: oak-core-spi: remove Guava from public API

* OAK-10247: oak-core-spi: remove Guava from public API - OSGi import cleanup

* OAK-10274: fix file mode

* OAK-10274: fix EOLs

* OAK-10274: fix EOLs

reschke committed:
OAK-10366: Update Oak trunk and Oak 1.22 to Jackrabbit 2.20.11 (#1036)

reschke committed:
OAK-10354: Release Oak 1.54.0 - fix oak-doc* versions

fabriziofortino committed:
OAK-10365: introduce mapping version in elastic index definition (#1035)

* OAK-10365: introduce mapping version in elastic index definition

* OAK-10365: minor improvements

anchela committed:
OAK-10318 : Improve AutoMembershipPrincipals#isInheritedMember

* OAK-10318 : Improve AutoMembershipPrincipals#isInheritedMember

* OAK-10318 : Improve AutoMembershipPrincipals#isInheritedMember (add trace to cycle-warning as suggested by joergH, fix typo, change order of checks)

reschke committed:
OAK-10322: oak-core: remove Guava from public API (#1037)

reschke committed:
OAK-10368: convert groovy sources to shaded guava (#1039)

reschke committed:
OAK-10323: remove all remaining references of native Guava (#1038)

* OAK-10323: remove all remaining references of native Guava

* OAK-10323: re-add Guava test dep to oak-store-composite

reschke committed:
OAK-10369: oak-segment-azure/oak-blob-cloud-azure leak Guava dependency - adjust import range (#1040)

dulceanu committed:
OAK-9922 - Parallel Compaction
Solved conflicts due to guava changes

dulceanu committed:
OAK-9922 - Parallel Compaction
Merge branch 'lweitzendorf-breadth-first-scan' into trunk

dulceanu committed:
Merge branch 'trunk' of github.com:apache/jackrabbit-oak into trunk

dulceanu committed:
Revert OAK-9922 - Parallel Compaction changes

kwin committed:
OAK-9837 Update to individual OSGi R7 dependencies (#1015)

* OAK-9837 Update to individual OSGi R7 dependencies

Update annotations to R7 as well

* Fix mocking for BundleContext.registerService(...) using Class as first
argument

* Fix OSGiIT for OSGI DS 1.4

Clarify in javadocs that service properties need String keys

* Update ConfigAdmin to version compliant with 1.6 in OSGi IT

Co-authored-by: Manfred Baedke <gre55877@adobe.com>

tihom88 committed:
OAK-10003: Upgrade tika to 1.28.5 (#758)

* OAK-10003: Upgrade tika to 1.28.5

* OAK-10003: Upgrade tika to 1.28.5

reschke committed:
OAK-10181: Flaky test: DataStoreBlobStoreStatsTest - ensure that the two created blobs are different (#1045)

reschke committed:
OAK-10316: Importer exceptions always should mention path - add missing SP character

thomasmueller committed:
OAK-301 Improve documentation

It was not clear what "insecure" means exactly. I tried to shorten the sentence a bit and make it clearer.

thomasmueller committed:
OAK-301 Improve documentation

Clarify what "reveal" means.

thomasmueller committed:
Merge pull request #1012 from apache/OAK-10333

OAK-10333 Improved logging for queries that that traverse 20'000 nodes

fabriziofortino committed:
OAK-10372: oak-search-elastic: similarity queries produce no relevant results (#1046)

* OAK-10372: oak-search-elastic: similarity queries produce no relevant results

* OAK-10372: (minor) improved comments

* OAK-10372: make similarity fields configurable

stefan-egli committed:
OAK-10347 : Adding small util class for exporting a tree as flat file (#1021)

* OAK-10347 : Adding small util class for exporting a tree as flat file

<truncated/>

---------

Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>

thomasmueller committed:
Reindexing with the tree store, plus some command line tools

tihom88 committed:
OAK-10381: DocumentStoreIndexerIT failure resolved (#1052)

mbaedke committed:
OAK-10367: update org.apache.sling.testing.osgi-mock dependency (#1044)

* OAK-10367: update org.apache.sling.testing.osgi-mock dependency

* OAK-10367: update org.apache.sling.testing.osgi-mock dependency

Reverted changes that are not related to o.a.sling.testing.osgi-mock.
Removed redundant explicit declaration of transitive dependencies of o.a.sling.testing.osgi-mock.junit4.

* OAK-10367: update org.apache.sling.testing.osgi-mock dependency

Added missing test dependencies.

---------

Co-authored-by: Manfred Baedke <gre55877@adobe.com>

mbaedke committed:
OAK-10379: DynamicGroupsTest#testCrossIDPMembership() fails occasionally (#1054)

Fixed timing issue.

Co-authored-by: Manfred Baedke <gre55877@adobe.com>

reschke committed:
OAK-10385: segment-azure: add javax.activation-api to test class path (#1055)

reschke committed:
OAK-10386: Bump netty dependency from 4.1.52.Final to 4.1.96.Final (#1056)

reschke committed:
OAK-10389: remove commons-lang dependency (#1058)

reschke committed:
OAK-10390: Update commons-io dependency to 2.13.0 (#1059)

* OAK-10390: Update commons-io dependency to 2.13.0

* OAK-10390: Update commons-io dependency to 2.13.0 - accept NPE for null arguments in filters (change in commons-io API)

reschke committed:
OAK-10391: improve TarFilesTest.testCollectBlobReferencesWithGenerationFilter diagnostics (#1060)

* OAK-10391: improve TarFilesTest.testCollectBlobReferencesWithGenerationFilter diagnostics

* OAK-10391: improve TarFilesTest.testCollectBlobReferencesWithGenerationFilter diagnostics - fix typo

reschke committed:
OAK-10392: Update (shaded) Guava to 32.1.2 (#1061)

reschke committed:
OAK-10394: Update commons-lang3 dependency to 3.13.0 (#1062)

reschke committed:
OAK-10395: Update commons-codec dependency to 1.16.0 (#1063)

* OAK-10390: Update commons-io dependency to 2.13.0

* OAK-10390: Update commons-io dependency to 2.13.0 - accept NPE for null arguments in filters (change in commons-io API)

* OAK-10395: Update commons-codec dependency to 1.16.0

reschke committed:
OAK-10396: segment-aws: remove unused commons-lang3 dependency (#1064)

reschke committed:
OAK-10397: oak-benchmarks/oak-it-osgi: update commons-compress dependency to 1.23.0 (#1065)

reschke committed:
OAK-10400: auth-ldap/benchmarks: use default dependency for commons-lang3 (#1067)

reschke committed:
OAK-10401: search-mt/solr-osgi: use default versions of commons-math3 (#1068)

jelmini committed:
OAK-10272 reverse OAK-10022

However, keep SegmentArchiveWriter#isRemote as it's now part of the SPI.

reschke committed:
OAK-10402: benchmarks: remove commons-math dependency (use math3 instead) (#1069)

reschke committed:
OAK-10403: update Tomcat JDBC dependency to 8.5.92 (#1072)

smiroslav committed:
Merge pull request #1070 from jelmini/OAK-10272_reverse_OAK-10022

OAK-10272 reverse OAK-10022

reschke committed:
OAK-10404: improve TarFilesTest.testCollectBlobReferences diagnostics (#1074)

thomasmueller committed:
Use a fixed directory (so we can use oak-run console and so on)

thomasmueller committed:
Support re-using the treeStore

reschke committed:
OAK-10405: document-store: potential NPE in AbstractTwoNodeTest teardown (#1076)

reschke committed:
OAK-10407: run/lucene: update gson dependencies to 2.10.1 (#1078)

thomasmueller committed:
Support reindexing the counter index

thomasmueller committed:
Support copying node stores

reschke committed:
OAK-10412: AbstractTwoNodeTest: check preconditions in setup (#1081)

reschke committed:
OAK-10415: AbstractTwoNodeTest: make teardown more robust (#1083)

reschke committed:
OAK-10416: AbstractTwoNodeTest: ensure RDB fixture uses empty DB (#1084)

reschke committed:
OAK-10414: RDB*Store: update Derby dependency to 10.15.2.0 (#1082)

mreutegg committed:
Merge pull request #1031 from mreutegg/OAK-10357

OAK-10357: Path option for documentstore-check

thomasmueller committed:
Merge branch 'apache:trunk' into OAK-10341-nuno

thomasmueller committed:
Tree store: configurable settings
